From ab03b8a6aae2c81c1678553d4f7d31c93b71ccb0 Mon Sep 17 00:00:00 2001
From: Girish Kumar M P <quic_gmp@quicinc.com>
Date: Sun, 13 Oct 2024 00:54:36 +0530
Subject: [PATCH] wifi: ath12k: Fix KW issue in ath12k driver

Fixed KW issues which are reported in ath12k driver
which might potentially cause crash.

Signed-off-by: Girish Kumar M P <quic_gmp@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/dp_rx.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/wireless/ath/ath12k/dp_rx.c b/drivers/net/wireless/ath/ath12k/dp_rx.c
index 13a13477..c81a9d7c 100644
--- a/drivers/net/wireless/ath/ath12k/dp_rx.c
+++ b/drivers/net/wireless/ath/ath12k/dp_rx.c
@@ -5637,6 +5637,9 @@ int ath12k_dp_rxdma_ring_sel_config_wcn7850(struct ath12k_base *ab)
 						       HAL_RXDMA_BUF,
 						       DP_RXDMA_REFILL_RING_SIZE,
 						       &tlv_filter);
+		if (ret)
+			break;
+
 	}
 
 	return ret;
-- 
2.34.1

